kzg/src/msm/msm_impls.rs

```diff
@@ -0,0 +1,70 @@
+use crate::{G1Affine, G1Fp, G1GetFp, G1ProjAddAffine, Scalar256, G1};
+use alloc::vec::Vec;
+
+#[cfg(feature = "arkmsm")]
+use super::arkmsm::arkmsm_msm::VariableBaseMSM;
+
+#[cfg(not(feature = "arkmsm"))]
+use super::tiling_pippenger_ops::tiling_pippenger;
+
+#[cfg(feature = "parallel")]
+use super::tiling_parallel_pippenger::{parallel_affine_conv, tiling_parallel_pippenger};
+
+#[cfg(feature = "parallel")]
+pub fn msm_parallel<
+    TG1: G1 + G1GetFp<TG1Fp>,
+    TG1Fp: G1Fp,
+    TG1Affine: G1Affine<TG1, TG1Fp>,
+    TProjAddAffine: G1ProjAddAffine<TG1, TG1Fp, TG1Affine>,
+>(
+    points: &[TG1Affine],
+    scalars: &[Scalar256],
+) -> TG1 {
+    tiling_parallel_pippenger(points, scalars)
+}
+
+pub fn msm_sequential<
+    TG1: G1 + G1GetFp<TG1Fp>,
+    TG1Fp: G1Fp,
+    TG1Affine: G1Affine<TG1, TG1Fp>,
+    TProjAddAffine: G1ProjAddAffine<TG1, TG1Fp, TG1Affine>,
+>(
+    points: &[TG1Affine],
+    scalars: &[Scalar256],
+) -> TG1 {
+    #[cfg(not(feature = "arkmsm"))]
+    {
+        tiling_pippenger(points, scalars)
```

Comment:
This function will panic if called with `points.len() == 1`:

```rs
let point = FsG1::rand();
let scalar = Scalar256::from_u8(&[0u8; 32]);
let result = msm::msm_impls::msm::<FsG1, FsFp, FsG1Affine, FsG1ProjAddAffine>(
    &[FsG1Affine::into_affine(&point)],
    &[scalar],
);
```

Reply:
Handled by the `< 8` check of the naive implementation.

Reply:
Maybe then add explicit `assert!(points.len() > 1)`? Now it's not obvoius that this function will panic

Reply:
added